pkgs/swift2objc/test/unit/filter_test.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:swift2objc/swift2objc.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Unit test for filter', () {
+    final thisDir = path.join(Directory.current.path, 'test/unit');
+
+    final file = path.join(thisDir, 'filter_test_input.swift');
+    final output = path.join(thisDir, 'filter_test_output.swift');
+      test(path.basename(file), () async {
+        final actualOutputFile = path.join(thisDir, path.basename(output));
+
+        await generateWrapper(Config(
+          input: FilesInputConfig(
+            files: [Uri.file(file)],
+          ),
+          outputFile: Uri.file(actualOutputFile),
+          tempDir: Directory(thisDir).uri,
+          preamble: '// Test preamble text',
+          filter: (declaration) => false,
```

Comment:
This test would be a bit better if you filtered some specific declarations. Try adding a few more classes, methods, globals, top level functions etc, and filter some of them but not all of them.

Reply:
Oh, also, can you add a method that is included by the filter, but refers to a class that is not included? What should happen in that case?